hnn_core/network.py

```diff
@@ -219,47 +293,443 @@ def __repr__(self):
+                numspikes is None or repeats is None):
+            raise ValueError('distribution, t0, sigma_t0, T, burst_f, '
+                             'burst_sigma_f, numspikes and repeats must all '
+                             'be defined')
+        drive = NetworkDrive()
+        drive['type'] = 'bursty'
+        drive['cell_specific'] = False
+        drive['seedcore'] = seedcore
+
+        drive['dynamics'] = dict(distribution=distribution, t0=t0,
+                                 sigma_t0=sigma_t0, T=T, burst_f=burst_f,
+                                 burst_sigma_f=burst_sigma_f,
+                                 numspikes=numspikes, repeats=repeats
+                                 )
+        drive['events'] = []
+
+        self._attach_drive(name, drive, weights_ampa, weights_nmda, location,
+                           space_constant, dispersion_time,
+                           cell_specific=False)
+
+    def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
+                      space_constant, dispersion_time, cell_specific=True):
+        """Attach a drive to network based on connectivity information
+
+        Parameters
+        ----------
+        name : str
+            Name of drive (must be unique)
+        drive : instance of NetworkDrive
+            Collection of parameters defining the dynamics of the drive
+        weights_ampa : dict or None
+            Synaptic weights of AMPA receptors on each targeted cell type
+            (keys). Weights can be zero or left omitted.
+        weights_nmda : dict or None
+            Synaptic weights of NMDA receptors on each targeted cell type
+            (keys). Weights can be zero or left omitted.
+        location : str
+            Target location of synapses ('distal' or 'proximal')
+        space_constant : float
+            Describes lateral dispersion (from column origin) of synaptic
+            delays within the simulated column
+        dispersion_time : dict
+            Synaptic delay at the column origin, dispersed laterally as a
+            function of the space_constant
+        cell_specific : bool
+            Whether each cell has unique connection parameters (default: True)
+            or all cells have common connections to a global (single) drive.
+
+        Attached drive is stored in self.external_drives[name]
+        self.pos_dict is updated, and self._update_gid_ranges() called
+        """
+        if name in self.external_drives:
+            raise ValueError(f"Drive {name} already defined")
+        if location not in ['distal', 'proximal']:
+            raise ValueError("Allowed drive target locations are: 'distal', "
+                             f"and 'proximal', got {location}")
+        # allow passing weights as None, but make iterable here
+        if weights_ampa is None:
+            weights_ampa = {}
+        if weights_nmda is None:
+            weights_nmda = {}
```

Comment:
none should be used for some sane defaults, not empty dict?

Reply:
I like being able to say `weights_nmda=None` (and that both default to `None`).

The reason for the block you've highlighted is to [do some logic tests on the inputs](https://github.com/cjayb/hnn-core/blob/5ea7f160b5bfdaae8fce715af54392d1e593f418/hnn_core/network.py#L583-L592)

I can't think of any sane defaults (other than none at all) for these weights...